examples/lua/nats_adapter.lua

```diff
@@ -0,0 +1,94 @@
+                        ["eventSource"] = "ceph:s3",
+                        ["awsRegion"] = request.ZoneGroup.Name,
+                        ["eventTime"] = request.Time,
+                        ["eventName"] = eventName,
+                        ["userIdentity"] = {
+                            ["principalId"] = request.User.Id
+						},
+                        ["requestParameters"] = {
+                            ["sourceIPAddress"] = ""
+						},
+                        ["responseElements"] = {
+                            ["x-amz-request-id"] =  request.Id,
+                            ["x-amz-id-2"] = request.RGWId
+						},
+                        ["s3"] = {
+                            ["s3SchemaVersion"] = "1.0",
+                            ["configurationId"] = configure,
+                            ["bucket"] = {
+                                ["name"] = request.Bucket.Name,
+                                ["ownerIdentity"] = {
+                                    ["principalId"] = request.Bucket.User.Id
+                                },
+                                ["arn"] = "arn:aws:s3:" .. request.ZoneGroup.Name .. "::" .. request.Bucket.Name,
+                                ["id"] = request.Bucket.Id
+						    },
+                            ["object"] = {
+                                ["key"] = request.Object.Name,
+                                ["size"] = request.Object.Size,
+                                ["eTag"] = "",
+                                ["versionId"] = request.Object.Instance,
+                                ["sequencer"] = string.format("%x", os.time()), 
+                                ["metadata"] ={
+				    request.HTTP.Metadata
+                                },
+                                ["tags"] = {
+                                    request.Tags
+                                }
+                            }
+                        },
+                        ["eventId"] = "",
+                        ["opaqueData"] = opaqueData
+                    }
+    }
+    return notification
+end
+
+supported_event = false
+configure = "mynotif1"
+opaqueData = "me@example.com"
+topic = "Bucket_Notification"
+
+--Obeject Created
+if Request.RGWOp == "put_obj" then
+    notification = toJson(Request ,'ObjectCreated:Put', opaqueData, configure)
+elseif Request.RGWOp == "post_obj" then
+    notification = toJson(Request ,'ObjectCreated:Post', opaqueData, configure)
+
+elseif Request.RGWOp == "copy_obj" then
+    notification = toJson(Request ,'ObjectCreated:Copy', opaqueData, configure)
+
+--Obeject Removed
+elseif Request.RGWOp == "delete_obj" then
+    notification = toJson(Request ,'ObjectRemoved:Delete', opaqueData, configure)
+end
+
+RGWDebugLog("number of metadata entires: " .. #Request.HTTP.Metadata)
+for k,v in pairs(Request.HTTP.Metadata) do
+    RGWDebugLog("key " .. k .. ", " .. "value=" .. v)
+end
+
+if supported_event == true then
```

Comment:
better to put the object creation code under this "if"

Reply:
also, we probably want to filter by a specific bucket name (as bucket notifications are usually per bucket)